scripts/webpack-utils/WatchCompiler.js

```diff
@@ -0,0 +1,171 @@
+      poll: process.env.WATCH_WITH_POLLING === 'true' && 2000,
+      ...watchOptions,
+    };
+
+    // Used to track whether or not to log a "change detected" message
+    this.isFirstCompile = true;
+    // the webpack watcher
+    // @see - https://webpack.js.org/api/node/#watching
+    this.watching = null;
+  }
+
+  /**
+   * @private
+   *
+   * Register additional compiler hooks.
+   *
+   * On `watchRun`, indicate that a change was detected if this isn't the first
+   * time the compilation has run.  Also, reset the timer to display how long
+   * this update compilation has been running.
+   *
+   * @returns {void}
+   */
+  initialize () {
+    super.initialize();
+
+    // Called any time a file is changed before compilation
+    // Called many times on the first compilation
+    this.compiler.hooks.watchRun.tap(this.name, () => {
+      console.log(`\n${this.name}:watch:watchRun`);
+
+      if (!this.isFirstCompile) {
+        console.log('');
+        console.log('Change detected, compiling...');
+        console.log('');
+      }
+
+      if (!this.timer) {
+        this.setTimer();
+      }
+    });
+
+    // Called when a compilation has finished
+    this.compiler.hooks.done.tap(this.name, (stats) => {
+      console.log(`${this.name}:watch:done`);
+
+      this.isFirstCompile = false;
+    });
+  }
+
+  /**
+   * `build` is not a supported method, so calling this will ALWAYS throw an
+   * error.  The caller must use the `watch` method.
+   *
+   * @returns {void}
+   */
+  build () {
+    throw new Error('`build` method not supported.  Use `watch` instead.');
+  }
+
+  /**
+   * @async
+   *
+   * Build and then Watch the files.
+   *
+   * @see - https://webpack.js.org/api/node/#watching
+   *
+   * @returns {Promise}
+   *   Resolves when the watcher has started without an error.
+   *   Rejects when the watcher encounters an error (NOT a stats error).
+   */
+  watch () {
+    return new Promise((resolve, reject) => {
+      this.watching = this.compiler.watch(this.watchOptions, (error, stats) => {
+        if (error) {
+          this.handleBuildError(error);
+          reject(error);
+          return;
+        }
+
+        resolve();
+      });
+    });
+  }
+
+  /**
+   * @async
+   *
+   * Stop the watcher.
+   *
+   * @returns {Promise}
+   *   Resolves when the watcher has been stopped / closed.
+   *   Rejects never.
+   */
+  stop () {
+    if (!this.watching) {
+      return Promise.resolve();
+    }
+
+    return new Promise((resolve, reject) => {
+      this.watching.close(() => {
+        // Note an error being passed to this callback is not documented in
+        // webpack's documentation
+        this.watching = null;
+        resolve();
+      });
+    });
+  }
+
+  /**
+   * To be called when the WatchCompiler is no longer needed.
+   *
+   * Stops the watcher, the compile timer, and dereferences the compiler.
+   *
+   * @returns {void}
```

Comment:
@brainthinks This returns a promise - please update.

Reply:
done